From 2f27d02f3208339ab0e96bb86c145bbd17bcc093 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:18 -0400
Subject: [ethernet] qlcnic: Allow any VLAN to be configured from VF

Message-id: <1408626422-3941-95-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88617
O-Subject: [PATCH RHEL 7.1 094/138] qlcnic: Allow any VLAN to be configured from VF.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 46428228b55df144f04cbad023907c177aa00d5f

o This patch reverts commit 1414abea048e0835c43600d62808ed8163897227
  (qlcnic: Restrict VF from configuring any VLAN mode.)
  This will allow same multicast address to be used with any VLAN
  instead of programming seperate (MAC, VLAN) tuples in adapter.
  This will help save some multicast filters.

Signed-off-by: Sucheta Chakraborty <sucheta.chakraborty@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_pf.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_pf.c
index 39dc957..bc85f99 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_pf.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_sriov_pf.c
@@ -13,8 +13,6 @@
 #define QLC_VF_MIN_TX_RATE 100
 #define QLC_VF_MAX_TX_RATE 9999
 #define QLC_MAC_OPCODE_MASK 0x7
-#define QLC_MAC_STAR_ADD 6
-#define QLC_MAC_STAR_DEL 7
 #define QLC_VF_FLOOD_BIT BIT_16
 #define QLC_FLOOD_MODE  0x5
 #define QLC_SRIOV_ALLOW_VLAN0 BIT_19
@@ -1203,13 +1201,6 @@ static int qlcnic_sriov_validate_cfg_macvlan(struct qlcnic_adapter *adapter,
  struct qlcnic_vport *vp = vf->vp;
  u8 op, new_op;
 
- if (((cmd->req.arg[1] & QLC_MAC_OPCODE_MASK) == QLC_MAC_STAR_ADD) ||
-     ((cmd->req.arg[1] & QLC_MAC_OPCODE_MASK) == QLC_MAC_STAR_DEL)) {
-  netdev_err(adapter->netdev, "MAC + any VLAN filter not allowed from VF %d\n",
-      vf->pci_func);
-  return -EINVAL;
- }
-
  if (!(cmd->req.arg[1] & BIT_8))
   return -EINVAL;
 
-- 
1.7.1